server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -489,8 +490,43 @@ public static GeoDistanceSortBuilder fromXContent(XContentParser parser, String
 
     @Override
     public SortFieldAndFormat build(QueryShardContext context) throws IOException {
+        GeoPoint[] localPoints = localPoints();
```

Comment:
These are all extracted into private methods to make building the BucketedSort simpler.